### DISCUSSION TITLE
.await syntax

### DISCUSSION THREAD

**Discussion by san127127** [dotnet/csharplang#4058] [General] [answered]
The current await syntax is not convenient when we want to chain the methods. For example,
```
(await GetSomeDataAsync()).DoSomething()
// Even worse when those methods are all async
await (await (await GetSomeDataAsync()).DoSomethingAsync())).DoSomethingElseAsync()
```

Maybe we can have something similar to rust lang's ```.await``` syntax, which is much clearer:
```
GetSomeDataAsync()
.await
.DoSomethingAsync()
.await
DoSomethingElseAsync()
.await
```

**Comment by canton7** [comment]
See:

 - The discussion in https://github.com/dotnet/csharplang/issues/35
 - https://github.com/dotnet/csharplang/discussions/1117

Another advantage is that this nicely supports a null-conditional await: `GetSomeDataAsync()?.await`. The disadvantage is that we now have two syntaxes which do exactly the same thing.

**Comment by HaloFour** [comment, marked as answer]
On request by a member of the design team I opened up this proposal which more or less mirrors your request: #4076
